### PR TITLE
fike ckpt stuff

### PR DESCRIPTION
**Pull request by samsja**

### CODE REVIEW

File: src/zeroband/checkpoint.py [hunk 1]
```diff
@@ -223,7 +223,12 @@ def save_shm(self) -> None:
             self.live_server.start_server()
         self._logger.info(f"Saved checkpoint to {ckpt_path} in {time.perf_counter() - time_start} seconds")
 
-    def save(self, ckpt_path: str, remote_ckpt_path: str | None, already_in_shm: bool = False) -> None:
+    def get_checkpoints_to_delete(ckpt_path: str, topk: int) -> list[str]:
+        checkpoints = [d for d in os.listdir(ckpt_path) if d.startswith("step_")]
+        sorted_checkpoints = sorted(checkpoints, key=lambda x: int(x.split("_")[1]), reverse=True)
+        return [os.path.join(ckpt_path, d) for d in sorted_checkpoints[topk:]]
+
```
Comment (Jackmin801) [inline src/zeroband/checkpoint.py:230]: This definition is never used? Guessing this is a duplicate from a previous attempt

**Review by Jackmin801 — APPROVED** [review]
Minor nit. Otherwise lgtm!

File: src/zeroband/checkpoint.py (hunk 1 above)
Comment (samsja) [reply src/zeroband/checkpoint.py:230]: indeed

Comment (Jackmin801) [inline src/zeroband/checkpoint.py:230]:
```suggestion
```

**Pull request closed without merging**